### ISSUE TITLE
Linux client crashes on xorg mouse button click #17

### ISSUE THREAD

**Issue by mfarley281** [ClassicUO/ClassicUO#1338]
O/S: KDE Neon 20.04 64-bit
CUO: Linux Native
Mouse: Logitech MX1000
Mouse button mapping command on login: 
xmodmap -e "pointer = 1 10 3 4 5 2 11 8 9 14 15 16 17 18 19 20 6 12 13 7 21 22 23 24"

My mouse has many buttons and I've remapped some of them using the command above (I'm not sure if the remapping plays a role in this bug).

With the remapping above, when I bump my mousewheel to the left, it normally acts as a "middle-click". All apps detect this as a middle-click.

When I open xev from a console and test the mouse buttons, the mousewheel-bump-left registers as **button #2** on the **downclick** event, and **button #17** on the **up/release** button event.

When I (usually accidentally) hit mousewheel-bump-left-button with the mouse over ClassicUO (not even logged in anywhere), the client crashes with the error below. I've just tested and confirm the exact timing of the crash: I**t occurs on the downclick event (button #2) of the mousewheel-bump-left action**, not the "up/release" event (button 17)
```
23:22:51 |  Error   | System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at (wrapper stelemref) System.Object.virt_stelemref_class_small_idepth(intptr,object)
  at ClassicUO.Game.Managers.UIManager.OnMouseButtonDown (ClassicUO.Input.MouseButtonType button) [0x00092] in <2138ac0977c346309e8635687da40cf5>:0 
  at ClassicUO.GameController.HandleSdlEvent (System.IntPtr userData, System.IntPtr ptr) [0x003fc] in <2138ac0977c346309e8635687da40cf5>:0 
  at (wrapper native-to-managed) ClassicUO.GameController.HandleSdlEvent(intptr,intptr)
  at (wrapper managed-to-native) SDL2.SDL.SDL_PollEvent(SDL2.SDL/SDL_Event&)
  at Microsoft.Xna.Framework.SDL2_FNAPlatform.PollEvents (Microsoft.Xna.Framework.Game game, Microsoft.Xna.Framework.Graphics.GraphicsAdapter& currentAdapter, System.Boolean[] textInputControlDown, System.Int32[] textInputControlRepeat, System.Boolean& textInputSuppress) [0x00606] in <27e296a293bb437d996ae17ffde5d055>:0 
  at (wrapper delegate-invoke) <Module>.invoke_void_Game_GraphicsAdapter&_bool[]_int[]_bool&(Microsoft.Xna.Framework.Game,Microsoft.Xna.Framework.Graphics.GraphicsAdapter&,bool[],int[],bool&)
  at Microsoft.Xna.Framework.Game.RunLoop () [0x00019] in <27e296a293bb437d996ae17ffde5d055>:0 
  at Microsoft.Xna.Framework.Game.Run () [0x00032] in <27e296a293bb437d996ae17ffde5d055>:0 
  at ClassicUO.Client.Run () [0x00045] in <2138ac0977c346309e8635687da40cf5>:0 
  at ClassicUO.Bootstrap.Main (System.String[] args) [0x002af] in <2138ac0977c346309e8635687da40cf5>:0 
exception inside UnhandledException handler: (null) assembly:/home/matt/Outlands/CUO_Client/ClassicUO/mscorlib.dll type:InvalidOperationException member:(null)

[ERROR] FATAL UNHANDLED EXCEPTION: System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at (wrapper stelemref) System.Object.virt_stelemref_class_small_idepth(intptr,object)
  at ClassicUO.Game.Managers.UIManager.OnMouseButtonDown (ClassicUO.Input.MouseButtonType button) [0x00092] in <2138ac0977c346309e8635687da40cf5>:0 
  at ClassicUO.GameController.HandleSdlEvent (System.IntPtr userData, System.IntPtr ptr) [0x003fc] in <2138ac0977c346309e8635687da40cf5>:0 
  at (wrapper native-to-managed) ClassicUO.GameController.HandleSdlEvent(intptr,intptr)
  at (wrapper managed-to-native) SDL2.SDL.SDL_PollEvent(SDL2.SDL/SDL_Event&)
  at Microsoft.Xna.Framework.SDL2_FNAPlatform.PollEvents (Microsoft.Xna.Framework.Game game, Microsoft.Xna.Framework.Graphics.GraphicsAdapter& currentAdapter, System.Boolean[] textInputControlDown, System.Int32[] textInputControlRepeat, System.Boolean& textInputSuppress) [0x00606] in <27e296a293bb437d996ae17ffde5d055>:0 
  at (wrapper delegate-invoke) <Module>.invoke_void_Game_GraphicsAdapter&_bool[]_int[]_bool&(Microsoft.Xna.Framework.Game,Microsoft.Xna.Framework.Graphics.GraphicsAdapter&,bool[],int[],bool&)
  at Microsoft.Xna.Framework.Game.RunLoop () [0x00019] in <27e296a293bb437d996ae17ffde5d055>:0 
  at Microsoft.Xna.Framework.Game.Run () [0x00032] in <27e296a293bb437d996ae17ffde5d055>:0 
  at ClassicUO.Client.Run () [0x00045] in <2138ac0977c346309e8635687da40cf5>:0 
  at ClassicUO.Bootstrap.Main (System.String[] args) [0x002af] in <2138ac0977c346309e8635687da40cf5>:0
```

**Comment by mfarley281** [comment]
`System.IndexOutOfRangeException: Index was outside the bounds of the array.`

I suspect CUO is not expecting such a high number for a mouse button (17 in this case)

**Comment by mfarley281** [comment]
Attaching client crash log
```
######################## [START LOG] ########################
ClassicUO [DEV_BUILD] - 0.1.8.19 - 04/26/2021 23:42:35
OS: Unix x64
Thread: CUO_MAIN_THREAD

Shard: play.uooutlands.com
ClientVersion: 7.0.15.1

Exception:
System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at (wrapper stelemref) System.Object.virt_stelemref_class_small_idepth(intptr,object)
  at ClassicUO.Game.Managers.UIManager.OnMouseButtonDown (ClassicUO.Input.MouseButtonType button) [0x00092] in <2138ac0977c346309e8635687da40cf5>:0 
  at ClassicUO.GameController.HandleSdlEvent (System.IntPtr userData, System.IntPtr ptr) [0x003fc] in <2138ac0977c346309e8635687da40cf5>:0 
  at (wrapper native-to-managed) ClassicUO.GameController.HandleSdlEvent(intptr,intptr)
  at (wrapper managed-to-native) SDL2.SDL.SDL_PollEvent(SDL2.SDL/SDL_Event&)
  at Microsoft.Xna.Framework.SDL2_FNAPlatform.PollEvents (Microsoft.Xna.Framework.Game game, Microsoft.Xna.Framework.Graphics.GraphicsAdapter& currentAdapter, System.Boolean[] textInputControlDown, System.Int32[] textInputControlRepeat, System.Boolean& textInputSuppress) [0x00606] in <27e296a293bb437d996ae17ffde5d055>:0 
  at (wrapper delegate-invoke) <Module>.invoke_void_Game_GraphicsAdapter&_bool[]_int[]_bool&(Microsoft.Xna.Framework.Game,Microsoft.Xna.Framework.Graphics.GraphicsAdapter&,bool[],int[],bool&)
  at Microsoft.Xna.Framework.Game.RunLoop () [0x00019] in <27e296a293bb437d996ae17ffde5d055>:0 
  at Microsoft.Xna.Framework.Game.Run () [0x00032] in <27e296a293bb437d996ae17ffde5d055>:0 
  at ClassicUO.Client.Run () [0x00045] in <2138ac0977c346309e8635687da40cf5>:0 
  at ClassicUO.Bootstrap.Main (System.String[] args) [0x002af] in <2138ac0977c346309e8635687da40cf5>:0 
######################## [END LOG] ########################




```

**Comment by mfarley281** [comment]
This is fixed in 0.1.8.20, thanks @andreakarasho !!
